### PR TITLE
Document Rock-on update and force deletion #274 #276.

### LINKED ISSUE #274: Document how to update rock-ons
How to update a given rock-on is a common question on our forum (as illustrated by https://forum.rockstor.com/t/rockons-app-update/3454).
Although some projects provide a way to update within the webUI or through a container restart, all rock-ons should support the "docker-way":
1. pull new image
2. stop container
3. remove container
4. re-create container
5. start container

Using Rockstor, this can be done by:
1. Stopping the rock-on
2. Uninstalling the rock-on
3. Installing the rock-on using the same settings

We should thus provide these instructions in our documentation, with proper mentions of any exceptions, if any.

### LINKED ISSUE #276: Force delete of a Rock-on
We have in place, for some time now, a command line script that does a force delete of a Rock-on by name via the command line:
```
/opt/rockstor/bin/delete-rockon 
Delete metadata, containers and images of a Rock-on
	Usage: /opt/rockstor/bin/delete-rockon <rockon name>
```
Although there are plans to integrate this forced/emergency rock-on removal method into our Web-UI, say via an advanced option / link possibly exposed after a set period of Rock-on transition post expected completion; or just in an advanced menu somewhere, it would still be useful to have this command line capability documented. Especially for older installs that pre-date our pending Web-UI improvements re dealing with problematic/stuck Rock-on install/un-install etc.

The proposed document page would be:
https://rockstor.com/docs/docker-based-rock-ons/overview.html

### PR DESCRIPTION
**Pull request by FroggyFlox**
Fixes #274 
Fixes #276 
@phillxnet, draft state for now until I verify proper formatting when ran under Apache2

### This pull request's proposal
This pull request (PR) proposes to document:
- the `delete-rockon` script that was previously not documented.
- how to uninstall a Rock-on.
- how to update a Rock-on from Rockstor's web-UI.

### Checklist
- [x] With the proposed changes no Sphinx errors or warnings are generated.
- [x] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

**Comment by FroggyFlox** [comment]
@phillxnet , all looks as intended under Apache2 so I'm moving to as "ready for review".

**Comment by phillxnet** [comment]
@FroggyFlox Well done. I'll take a look later today and publish shortly there after likely as I have a much smaller pending change in another area in the works.

File: interface/overview.rst [hunk 1]
```diff
@@ -416,3 +416,79 @@ modifications made to the rock-on, Rockstor will apply the new settings on the
...
+latter will remain intact. If you re-used the same *Shares* for the Rock-on as
+before its uninstallation, you should thus be back in the same state as before,
+but with the latest Docker image available.
+
+Note, however, that some Rock-ons use pinned versions for their underlying
+image(s); these therefore cannot be updated. These include the *PostgreSQL 9.5*,
+*PostgreSQL 10.6*, and *YouTrack official* Rock-ons.
+
+.. note::
+    Although some of the applications ran inside Rock-ons have an "Update"
```
Comment (phillxnet) [inline interface/overview.rst:487]:
```suggestion
    Although some of the applications inside Rock-ons have an "Update"
```
Not a show stopper but if you fancy accepting then it reads a little nicer to me.

**Review by phillxnet — CHANGES_REQUESTED** [review]
@FroggyFlox This is an excellent and timely addition. Thanks and much appreciated.

Only one minor soft suggestion. Let me know either way and I'll merge and publish later today.

**Comment by FroggyFlox** [comment]
Thanks for the edits, @phillxnet , it does read a lot better that way.

**Comment by phillxnet** [comment]
Pr product in production.

**Pull request merged**